File: js/src/maplestory/api/dto/character/characterPopularityDto.ts
import { CharacterPopularityDtoBody } from '@src/maplestory/api/response/character/characterPopularityDtoBody';

/**
 * 캐릭터 인기도 정보
 */
class CharacterPopularityDto {
  /**
   * 조회 기준일
   */
  date: Date | null;

  /**
   * 캐릭터 인기도
   */
  popularity: number;

  constructor(obj: CharacterPopularityDtoBody) {
    const { date, popularity } = obj;

    this.date = date ? new Date(date) : null;
    this.popularity = popularity;
  }
}

export { CharacterPopularityDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterPopularityDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.ZonedDateTime;

/**
 * 角色名聲資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterPopularityDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterPopularityDTO {

    /**
     * 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
     */
    @JsonProperty("date")
    private String date;

    /**
     * 角色名聲
     */
    @JsonProperty("popularity")
    private long popularity;

    /**
     * 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
     */
    public ZonedDateTime getDate() {
        return date != null
                ? Utils.toZonedDateTime(this.date)
                : null;
    }
}
